### PR TITLE
Fix maddie layout for ploopy/trackball_nano

### PR DESCRIPTION
**Pull request by aidalgol**
## Description
This keymap was using a user function that had been removed from QMK, so it was never being called.  The function that replaces it provides a slightly different interface, so the implementation in this keymap had to be updated as well as the
function signature.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
@hexadecatrienoic will need to sign off on the change per contribution guidelines.

File: keyboards/ploopyco/trackball_nano/keymaps/maddie/keymap.c [hunk 1]
```diff
@@ -18,39 +18,24 @@
...
-		if (delta_y > 15) {
-			mouse_report->v = -1;
-			delta_y = 0;
-		} else if (delta_y < -15) {
-			mouse_report->v = 1;
-			delta_y = 0;
-		}
-    } else {
-        mouse_report->x = x;
-		mouse_report->y = y;
```
Comment (hexadecatrienoic) [inline keyboards/ploopyco/trackball_nano/keymaps/maddie/keymap.c:52]: the `delta_` code should remain - this is to reduce the number of scroll events to a usable level.

**Review by hexadecatrienoic — CHANGES_REQUESTED** [review]
thank you for updating this!

File: keyboards/ploopyco/trackball_nano/keymaps/maddie/keymap.c (hunk 1 above)
Comment (aidalgol) [reply keyboards/ploopyco/trackball_nano/keymaps/maddie/keymap.c:52]: When I had the `delta_` code in there, I would get small but visible mouse movement along with scrolling, so I think it needs to be modified, but I'm not sure how exactly.

Comment (aidalgol) [reply keyboards/ploopyco/trackball_nano/keymaps/maddie/keymap.c:52]: @hexadecatrienoic I managed to figure out how to restore the `delta_` logic while eliminating the mouse movement I was seeing.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by dhorlick** [comment]
Pretty sure this doesn't compile…
```
dhorlick@brogmoid:~/diy/build/qmk_firmware$ qmk compile -kb ploopyco/trackball_nano -km maddie
⋮
keyboards/ploopyco/trackball_nano/keymaps/maddie/keymap.c:35:9: error: ‘delta_x’ undeclared (first use in this function)
         delta_x += mouse_report.x;
         ^
```

**Pull request merged**